src/main/java/com/github/princesslana/eriscasper/repository/event/GuildsFromEvents.java

```diff
@@ -36,106 +96,43 @@ public GuildsFromEvents(Observable<Event> eventObservable) {
-                                      .guildId(guildId)
-                                      .build()));
-                  return ImmutableMap.copyOf(mutableMap);
-                });
-    Observable<ImmutableMap<Snowflake, Channel>> guildDeleteWithChannelsListener =
-        eventObservable
-            .ofType(GuildDeleteEvent.class)
-            .map(GuildDeleteEvent::unwrap)
-            .scan(
-                emptyChannelMap,
-                (map, guild) -> {
-                  Map<Snowflake, Channel> mutableMap = new HashMap<>(map);
-                  Snowflake guildId = guild.getId();
-                  mutableMap
-                      .values()
-                      .stream()
-                      .filter(channel -> guildId.equals(channel.getGuildId().orElse(null)))
-                      .map(Channel::getId)
-                      .forEach(mutableMap::remove);
-                  return ImmutableMap.copyOf(mutableMap);
-                });
-    Observable<ImmutableMap<Snowflake, Channel>> channelCreateListener =
-        eventObservable
-            .ofType(ChannelCreateEvent.class)
-            .map(ChannelCreateEvent::unwrap)
-            .scan(
-                emptyChannelMap,
-                (map, channel) -> {
-                  Map<Snowflake, Channel> mutableMap = new HashMap<>(map);
-                  mutableMap.put(channel.getId(), channel);
-                  return ImmutableMap.copyOf(mutableMap);
-                });
-    Observable<ImmutableMap<Snowflake, Channel>> channelDeleteListener =
-        eventObservable
-            .ofType(ChannelDeleteEvent.class)
-            .map(ChannelDeleteEvent::unwrap)
-            .scan(
-                emptyChannelMap,
-                (map, channel) -> {
-                  Map<Snowflake, Channel> mutableMap = new HashMap<>(map);
-                  mutableMap.remove(channel.getId());
-                  return ImmutableMap.<Snowflake, Channel>builder().putAll(mutableMap).build();
-                });
-
-    return guildCreateWithChannelsListener
-        .mergeWith(guildDeleteWithChannelsListener)
-        .mergeWith(channelCreateListener)
-        .mergeWith(channelDeleteListener)
+    return generateGuildCreateListener()
+        .map(ADD_GUILD_CHANNELS_FUNCTION)
+        .mergeWith(generateGuildDeleteListener().map(REMOVE_GUILD_CHANNELS_FUNCTION))
+        .mergeWith(
+            eventObservable
+                .ofType(ChannelCreateEvent.class)
+                .map(ChannelCreateEvent::unwrap)
+                .map(ADD_CHANNEL_FUNCTION))
+        .mergeWith(
+            eventObservable
+                .ofType(ChannelDeleteEvent.class)
+                .map(ChannelDeleteEvent::unwrap)
+                .map(REMOVE_CHANNEL_FUNCTION))
+        .scan(ImmutableMap.<Snowflake, Channel>of(), (map, function) -> function.apply(map))
+        .doOnError(Throwable::printStackTrace)
         .replay(1);
   }
 
+  // Used due to avoiding redundant code
```

Comment:
redundant comment :)

Reply:
I added in code to the guild and channel test to test what the bug report pointed out. If you'd like I can add it in, also I'll remove the second comment